### ISSUE TITLE
www-client/ungoogled-chromium-89.0.4389.90-r1: Unsupported instrumentation profile format version with USE=pgo and clang/LLVM 11.1.0

### ISSUE THREAD

**Issue by 0xC0ncord** [PF4Public/gentoo-overlay#80]
I am building ungoogled-chromium with `USE=pgo` and a clang/LLVM 11.1.0 toolchain. I get these errors very early during compilation:
```
error: Could not read profile ../../chrome/build/pgo_profiles/chrome-linux-4389-1615484746-845f8166c73d01aac66fc676e3122d92508416a9.profdata: Unsupported instrumentation profile format version
```
Without `USE=pgo`, it builds successfully. My assumption is that the pgo profiles currently shipped in chromium/UGO are not compatible with clang/LLVM 11.1.0.

`emerge -pqv ungoogled-chromium`
```
[ebuild   R   ] www-client/ungoogled-chromium-89.0.4389.90-r1  USE="cfi clang closure-compile convert-dict cups hangouts official optimize-thinlto optimize-webui pgo* proprietary-codecs pulseaudio screencast (selinux) suid system-ffmpeg system-harfbuzz system-icu system-jsoncpp system-libevent system-openh264 system-openjpeg system-re2 tcmalloc thinlto vaapi vdpau wayland -custom-cflags -enable-driver -headless -kerberos -system-libvpx -widevine" L10N="-am -ar -bg -bn -ca -cs -da -de -el -en-GB -es -es-419 -et -fa -fi -fil -fr -gu -he -hi -hr -hu -id -it -ja -kn -ko -lt -lv -ml -mr -ms -nb -nl -pl -pt-BR -pt-PT -ro -ru -sk -sl -sr -sv -sw -ta -te -th -tr -uk -vi -zh-CN -zh-TW"
```

`emerge --info`
https://ghostbin.co/paste/j64cf

`build.log`
https://ghostbin.co/paste/ythv

**Comment by perfect7gentleman** [comment]
Jusy disable PGO. Afaik, it's upstream bug.

**Comment by PF4Public** [comment]
It is true, that `pgo` does not function and it is a mystery to me, how did Google create this profdata. They could be using llvm-12 or 13 or be simply fooling us :)

I should probably mask this use flag temporarily or idk. Honestly saying, I wanted to do a RTFM on how it is generally handled in Gentoo, but didn't have the time, so left it as is.

**Comment by perfect7gentleman** [comment]
From Chromium ebuild
># Disable PGO, because profile data is only compatible with >=clang-11

**Comment by 0xC0ncord** [comment]
Strange, considering it had worked for me previously. Though, that was likely when I was still using clang 11.0.0. I have no issue closing this as I'm unsure of the actual performance benefits of using PGO. That being said, I think it would be best to include a warning about `USE=pgo` instead of just outright disabling it. Gentoo is all about choice after all!

**Comment by PF4Public** [comment]
> Strange, considering it had worked for me previously.

It did work! Up untill 89. I did myself compile [ungoogled-chromium-bin](https://github.com/PF4Public/ungoogled-chromium-bin/releases) with `pgo` before.

**Comment by r7l** [comment]
It just compiled for me with PGO, using the 12.0.0_rc3 packages of llvm / clang. I am writing this comment from it. I know the packages are masked and considered unstable but it all went through smoothly. I might have to switch back and forth with my clang environment for the time being as i don't want to use it this unstable version for any other package.

I had the same error eariler on with clang 11. So i thought about giving it a try and it actually seems as if newer Chromium builds require version 12+.

**Comment by PF4Public** [comment]
@r7l That is a good find! I think, I'll add a conditional for `pgo` USE flag then.

**Comment by PF4Public** [comment]
> I'm unsure of the actual performance benefits of using PGO

@0xC0ncord I've just rebuilt chromium with PGO. I didn't do a rigorous test. I just did some clicking, tab switching, settings opening, some clicking on a JS-heavy website, reinstalled chromium immediately from binpkg and repeated the clicking. Tentatively it feels a small fraction of a second faster. Thus I think, there must be some benefits of using PGO.
